File: demos/squad-room/dotnet/Activity.cs
using System.Text.Json;

namespace SquadRoomDemo;

/// <summary>Command Line activity panel — squad-room team variant. Mirrors python/activity.py exactly.</summary>
public static class Activity
{
    const string Ink = "\u001b[1m", Faint = "\u001b[2m", Reset = "\u001b[0m", Box = "\u001b[7m";
    static string B(string label) => $"{Box} {label} {Reset}";

    public static void Header(string title)
    {
        Console.WriteLine($"{Ink}{title}{Reset}");
        Console.WriteLine($"{Faint}// a whole AI team shares one governed Toolbox — watch every member hit the same box{Reset}");
    }

    public static void Member(string name, string msg) =>
        Console.WriteLine($"\n{Ink}>_ [{name}]{Reset} \"{msg}\"");

    public static void ToolSearch(string query, int roundTrip) =>
        Console.WriteLine($"   {B("tool_search")} query: \"{query}\"   {Faint}// round-trip #{roundTrip}{Reset}");

    public static void Matched(IEnumerable<string> names) =>
        Console.WriteLine($"   {B("matched")} {string.Join(", ", names)}");

    public static void CallTool(string name, object args) =>
        Console.WriteLine($"   {B("call_tool")} {name}({JsonSerializer.Serialize(args)})");

    public static void Result(string text) => Console.WriteLine($"   {Faint}=> {text}{Reset}");

    public static void Counter(int boxSize, int contextSize, int members)
    {
        Console.WriteLine($"\n{B($"tools in the shared box: {boxSize}")}   ·   {B($"tools in each member's context: {contextSize}")}");
        Console.WriteLine($"{Faint}// {members} team members, one governed box. each sees only the meta-tools. flat cost for the whole team.{Reset}");
    }
}

File: demos/squad-room/python/activity.py
"""Command Line activity panel — squad-room team variant. Mirrors dotnet/Activity.cs exactly."""

INK = "\033[1m"; FAINT = "\033[2m"; RESET = "\033[0m"; BOX = "\033[7m"


def _box(label):
    return f"{BOX} {label} {RESET}"


def member(name, msg):
    print(f"\n{INK}>_ [{name}]{RESET} \"{msg}\"")


def tool_search(query, round_trip):
    print(f"   {_box('tool_search')} query: \"{query}\"   {FAINT}// round-trip #{round_trip}{RESET}")


def matched(names):
    print(f"   {_box('matched')} {', '.join(names) if names else '(none)'}")


def call_tool(name, arguments):
    print(f"   {_box('call_tool')} {name}({arguments})")


def result(text):
    print(f"   {FAINT}=> {text}{RESET}")


def counter(box_size, context_size, members):
    print(f"\n{BOX} tools in the shared box: {box_size} {RESET}   ·   "
          f"{BOX} tools in each member's context: {context_size} {RESET}")
    print(f"{FAINT}// {members} team members, one governed box. each sees only the meta-tools. "
          f"flat cost for the whole team.{RESET}")


def header(title):
    print(f"{INK}{title}{RESET}")
    print(f"{FAINT}// a whole AI team shares one governed Toolbox — watch every member hit the same box{RESET}")
